### PR TITLE
Mu kafka management service

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

File: modules/kafka/src/test/scala/higherkindness/mu/rpc/kafka/ServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  test("list topics") {
+    withKafka { settings: AdminClientSettings[IO] =>
+      withClient(settings) { client =>
+        for {
+          list <- client.listTopics(Empty).attempt
+          _    <- IO(println(list))
+          _    <- IO(assert(list.isRight))
+        } yield ()
+      }.unsafeRunSync()
```
[the remark this answers is not in the document]
Comment (BenFradet) [reply modules/kafka/src/test/scala/higherkindness/mu/rpc/kafka/ServiceSpec.scala]: I was using `List` in request response types, that's why it was failing

**Comment by BenFradet** [comment]
I currently have an issue with `Map`s when they are in responses.

I had a response which was:

```scala
final case class Topics(topics: Map[String, TopicDescription])
```

and caused:

`io.grpc.StatusRuntimeException: CANCELLED: call already cancelled`

turning it into

```scala
final case class Topics(topics: List[TopicDescription])
```

made the problem disappear.

**Comment by juanpedromoreno** [comment]
That's weird, AFAIK avro4s is managing the `Map`'s serialization correctly :/

**Comment by BenFradet** [comment]
List of tuples do not seem to work either with the same issue as maps.

**Comment by BenFradet** [comment]
I think it's good to review now

File: modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/kafkaManagementService.scala [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+  @service(Protobuf)
+  trait KafkaManagement[F[_]] {
+    def alterConfigs(acr: AlterConfigsRequest): F[Unit]
+    def createPartitions(cpr: CreatePartitionsRequest): F[Unit]
+    def createTopic(ctr: CreateTopicRequest): F[Unit]
+    def createTopics(ctrs: CreateTopicRequests): F[Unit]
+    def deleteTopic(t: String): F[Unit]
+    def deleteTopics(ts: DeleteTopicsRequest): F[Unit]
+    def describeCluster(r: Empty.type): F[Cluster]
```
Comment (tzimisce012) [inline modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/kafkaManagementService.scala]: Why do you need the parameter empty?

File: modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/KafkaManagementImpl.scala [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import fs2.kafka._
+import higherkindness.mu.rpc.protocol.Empty
+import org.apache.kafka.clients.admin.{NewPartitions, NewTopic}
+
+import kafkaManagementService._
+
+class KafkaManagementImpl[F[_]: ContextShift: Concurrent] private[kafka] (
+    adminClient: KafkaAdminClient[F]
+) extends KafkaManagement[F] {
+  override def alterConfigs(acr: AlterConfigsRequest): F[Unit] =
```
Comment (tzimisce012) [inline modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/KafkaManagementImpl.scala]: lint: put a blank line in between :p

File: project/ProjectPlugin.scala [hunk 4]
```diff
@@ -24,38 +24,40 @@ object ProjectPlugin extends AutoPlugin {
...
+      val scopt: String              = "3.7.1"
+      val scalacheck: String         = "1.14.2"
+      val scalacheckToolbox: String  = "0.3.1"
+      val scalamock: String          = "4.4.0"
+      val scalatest: String          = "3.0.8"
+      val skeuomorph: String         = "0.0.15"
+      val slf4j: String              = "1.7.28"
+      val dropwizard: String         = "4.1.0"
+      val fs2Kafka: String           = "0.20.1"
+      val embeddedKafka: String      = "2.3.0"
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: Could we keep the alphabetical order?

File: modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/kafkaManagementService.scala [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    def listConsumerGroups(r: Empty.type): F[ConsumerGroupListings]
+    def listTopics(r: Empty.type): F[TopicListings]
+  }
+
+  object KafkaManagement {
+    def buildInstance[F[_]: ContextShift: Concurrent](
+        settings: AdminClientSettings[F]
+    ): Resource[F, KafkaManagement[F]] =
+      adminClientResource[F](settings)
+        .map(c => new KafkaManagementImpl(c))
```
Comment (fedefernandez) [inline modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/kafkaManagementService.scala]:
```suggestion
        .map(new KafkaManagementImpl(_))
```

File: modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/kafkaManagementService.scala (hunk 2 above)
Comment (BenFradet) [reply modules/kafka/src/main/scala/higherkindness/mu/rpc/kafka/kafkaManagementService.scala]: I think this is a grpc limitation.

**Pull request merged**